DisguiseUnityRenderStream/Runtime/PipelineAbstraction/CameraCapture.cs

```diff
@@ -0,0 +1,299 @@
+            {
+                m_cameraTexture = cameraTexture;
+                m_depthTexture = depthTexture;
+            }
+            
+            /// <summary>
+            /// Refers to <see cref="CameraCapture.cameraTexture"/>.
+            /// </summary>
+            public RenderTexture cameraTexture => m_cameraTexture;
+        
+            /// <summary>
+            /// Refers to <see cref="CameraCapture.depthTexture"/>.
+            /// </summary>
+            public RenderTexture depthTexture => m_depthTexture;
+        }
+
+        /// <summary>
+        /// Called once textures have been captured and are ready to be consumed.
+        /// </summary>
+        public event Action<ScriptableRenderContext, Capture> onCapture = delegate {};
+        
+        /// <summary>
+        /// <para>
+        /// The captured color texture. Its format is defined by <see cref="description"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// Note: while the texture may be configured for MSAA, Unity will always use an implicitly resolved version
+        /// of the texture for APIs such <see cref="CommandBuffer.CopyTexture(RenderTargetIdentifier, RenderTargetIdentifier)"/>
+        /// and see <see cref="Texture.GetNativeTexturePtr"/>.
+        /// </para>
+        ///
+        /// <remarks>
+        /// To avoid unnecessary texture copies, this refers directly to <see cref="Camera.targetTexture"/>.
+        /// </remarks>
+        /// </summary>
+        public RenderTexture cameraTexture => m_cameraTexture;
+        
+        /// <summary>
+        /// The captured depth texture. Its format is defined by <see cref="description"/>.
+        /// <remarks><see langword="null"/> if <see cref="description"/> is not configured for depth.</remarks>
+        /// </summary>
+        public RenderTexture depthTexture => m_depthTexture;
+
+        /// <summary>
+        /// Defines the color and depth textures. The textures are automatically disposed and created on change.
+        /// </summary>
+        public CameraCaptureDescription description
+        {
+            get => m_description;
+            set
+            {
+                if (m_description != value)
+                {
+                    m_description = value;
+                    Refresh();
+                }
+            }
+        }
+
+        [SerializeField]
+        CameraCaptureDescription m_description = CameraCaptureDescription.Default;
+
+        DepthCopy m_depthCopy;
+        Camera m_camera;
+        RenderTexture m_cameraTexture;
+        RenderTexture m_depthTexture;
+        
+#if DEBUG
+        bool m_HasSetSecondNames;        
+#endif
+
+#if UNITY_EDITOR
+        CameraCaptureDescription m_lastDescription = CameraCaptureDescription.Default;
+        bool m_EditorChangedFlag; // Ensures thread safety (OnValidate is on a different thread)
+        
+        void OnValidate()
+        {
+            m_EditorChangedFlag = true;
```

Comment:
I don't think this flag is strictly necessary any more. It only adds confusion for a negligible (debatable?) performance gain, IMO.